lib/travis/sso/generic.rb

```diff
@@ -115,6 +115,8 @@ def login(request)
           user = data['user'].merge('token' =>  token)
 
           if authorized?(user)
+            provider = get_provider(user).empty? ? 'github' : get_provider(user)
+            template(login_page, request, provider: provider)
```

Comment:
you need to update the login_page here and set proper provider - it's used later in handshake